File: moyai_app/lib/explore/model/entities/option_for_access.dart
class OptionForAccess {
  final String accessId;
  final String? rampWidth;
  final String? rampLength;
  final String? imageOfRamp;
  final List<String> railsOnRamp;
  final List<String> railsOnSteps;
  final List<String> rampFeatures;
  final String? numberOfSteps;
  final List<String> rampForAccess;
  final String? entranceDoorCm;
  final String? otherObstacles;
  final List<String> featuresOfSteps;
  final List<String> optionsForAccessDescription;
  final List<String> entranceFeatures;
  final List<String> liftForThisAccess;
  final List<String> signageAndDirection;
  final List<String> stepsForThisAccess;
  final String? rampDegreeMeasurement;
  final List<String> escalatorForThisAccess;
  final List<String> obstaclesWithTheAccess;
  final String? imageOfStairsForThisAccess;
  final bool? isThereObstaclesWithThisAccess;
  final String createdAt;
  final String updatedAt;

  OptionForAccess({
    required this.accessId,
    this.rampWidth,
    this.rampLength,
    this.imageOfRamp,
    required this.railsOnRamp,
    required this.railsOnSteps,
    required this.rampFeatures,
    this.numberOfSteps,
    required this.rampForAccess,
    this.entranceDoorCm,
    this.otherObstacles,
    required this.featuresOfSteps,
    required this.optionsForAccessDescription,
    required this.entranceFeatures,
    required this.liftForThisAccess,
    required this.signageAndDirection,
    required this.stepsForThisAccess,
    this.rampDegreeMeasurement,
    required this.escalatorForThisAccess,
    required this.obstaclesWithTheAccess,
    this.imageOfStairsForThisAccess,
    this.isThereObstaclesWithThisAccess,
    required this.createdAt,
    required this.updatedAt,
  });

  factory OptionForAccess.fromJson(Map<String, dynamic> json) {
    return OptionForAccess(
      accessId: json['accessId'],
      rampWidth: json['rampWidth'],
      rampLength: json['rampLength'],
      imageOfRamp: json['imageOfRamp'],
      railsOnRamp: List<String>.from(json['railsOnRamp'] ?? []),
      railsOnSteps: List<String>.from(json['railsOnSteps'] ?? []),
      rampFeatures: List<String>.from(json['rampFeatures'] ?? []),
      numberOfSteps: json['numberOfSteps'],
      rampForAccess: List<String>.from(json['rampForAccess'] ?? []),
      entranceDoorCm: json['entranceDoorCm'],
      otherObstacles: json['otherObstacles'],
      featuresOfSteps: List<String>.from(json['featuresOfSteps'] ?? []),
      optionsForAccessDescription: List<String>.from(json['optionsForAccessDescription'] ?? []),
      entranceFeatures: List<String>.from(json['entranceFeatures'] ?? []),
      liftForThisAccess: List<String>.from(json['liftForThisAccess'] ?? []),
      signageAndDirection: List<String>.from(json['signageAndDirection'] ?? []),
      stepsForThisAccess: List<String>.from(json['stepsForThisAccess'] ?? []),
      rampDegreeMeasurement: json['rampDegreeMeasurement'],
      escalatorForThisAccess: List<String>.from(json['escalatorForThisAccess'] ?? []),
      obstaclesWithTheAccess: List<String>.from(json['obstaclesWithTheAccess'] ?? []),
      imageOfStairsForThisAccess: json['imageOfStairsForThisAccess'],
      isThereObstaclesWithThisAccess: json['isThereObstaclesWithThisAccess'],
      createdAt: json['createdAt'],
      updatedAt: json['updatedAt'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'accessId': accessId,
      'rampWidth': rampWidth,
      'rampLength': rampLength,
      'imageOfRamp': imageOfRamp,
      'railsOnRamp': railsOnRamp,
      'railsOnSteps': railsOnSteps,
      'rampFeatures': rampFeatures,
      'numberOfSteps': numberOfSteps,
      'rampForAccess': rampForAccess,
      'entranceDoorCm': entranceDoorCm,
      'otherObstacles': otherObstacles,
      'featuresOfSteps': featuresOfSteps,
      'optionsForAccessDescription': optionsForAccessDescription,
      'entranceFeatures': entranceFeatures,
      'liftForThisAccess': liftForThisAccess,
      'signageAndDirection': signageAndDirection,
      'stepsForThisAccess': stepsForThisAccess,
      'rampDegreeMeasurement': rampDegreeMeasurement,
      'escalatorForThisAccess': escalatorForThisAccess,
      'obstaclesWithTheAccess': obstaclesWithTheAccess,
      'imageOfStairsForThisAccess': imageOfStairsForThisAccess,
      'isThereObstaclesWithThisAccess': isThereObstaclesWithThisAccess,
      'createdAt': createdAt,
      'updatedAt': updatedAt,
    };
  }

  static List<OptionForAccess> fromJsonList(List<dynamic> jsonList) {
    return jsonList.map((json) => OptionForAccess.fromJson(json)).toList();
  }

  static List<Map<String, dynamic>> toJsonList(List<OptionForAccess> options) {
    return options.map((option) => option.toJson()).toList();
  }
}

File: backend-code/src/db/entities/OptionForAccess.ts
import {
  Entity,
  BaseEntity,
  Column,
  PrimaryGeneratedColumn,
  CreateDateColumn,
  UpdateDateColumn,
  ManyToOne,
  JoinColumn,
  Relation,
} from "typeorm";
import { Venue } from "./Venue.js";

@Entity("option_for_access")
export class OptionForAccess extends BaseEntity {
  @PrimaryGeneratedColumn("uuid")
  accessId!: string;

  @ManyToOne(() => Venue, (venue) => venue.optionsForAccess)
  @JoinColumn({ name: "venue_id" })
  venue!: Relation<Venue>;

  @Column({ name: "ramp_width", nullable: true })
  rampWidth?: string;

  @Column({ name: "ramp_length", nullable: true })
  rampLength?: string;

  @Column({ name: "image_of_ramp", nullable: true })
  imageOfRamp?: string;

  @Column("simple-array", { name: "rails_on_ramp", nullable: true })
  railsOnRamp?: string[];

  @Column("simple-array", { name: "rails_on_steps", nullable: true })
  railsOnSteps?: string[];

  @Column("simple-array", { name: "ramp_features", nullable: true })
  rampFeatures?: string[];

  @Column({ name: "number_of_steps", nullable: true })
  numberOfSteps?: string;

  @Column("simple-array", { name: "ramp_for_access", nullable: true })
  rampForAccess?: string[];

  @Column({ name: "entrance_door_cm", nullable: true })
  entranceDoorCm?: string;

  @Column({ name: "other_obstacles", nullable: true })
  otherObstacles?: string;

  @Column("simple-array", { name: "features_of_steps", nullable: true })
  featuresOfSteps?: string[];

  @Column("simple-array", { name: "option_for_access", nullable: true })
  optionForAccess?: string[];

  @Column("simple-array", { name: "entrance_features", nullable: true })
  entranceFeatures?: string[];

  @Column("simple-array", { name: "lift_for_this_access", nullable: true })
  liftForThisAccess?: string[];

  @Column("simple-array", { name: "sinage_and_direction", nullable: true })
  sinageAndDirection?: string[];

  @Column("simple-array", { name: "steps_for_this_access", nullable: true })
  stepsForThisAccess?: string[];

  @Column({ name: "ramp_degree_measurement", nullable: true })
  rampDegreeMeasurement?: string;

  @Column("simple-array", { name: "escalator_for_this_access", nullable: true })
  escalatorForThisAccess?: string[];

  @Column("simple-array", { name: "obstacles_with_the_access", nullable: true })
  obstaclesWithTheAccess?: string[];

  @Column({ name: "image_of_stairs_for_this_access", nullable: true })
  imageOfStairsForThisAccess?: string;

  @Column("simple-array", {
    name: "is_there_obstacles_with_this_access",
    nullable: true,
  })
  isThereObstaclesWithThisAccess?: string[];

  @CreateDateColumn({ name: "created_at", type: "timestamptz" })
  createdAt!: Date;

  @UpdateDateColumn({ name: "updated_at", type: "timestamptz" })
  updatedAt!: Date;
}
